database.hh

```diff
@@ -364,6 +364,7 @@ struct table_stats {
 class table : public enable_lw_shared_from_this<table> {
 public:
     struct config {
+        const dht::sharding_info& sharding_info;
```

Comment:
It felt more appropriate to pass sharding_info down from database to table instead of putting it into the schema. I agree it would be simpler but it will create a coupling that is not justified in any way. schema and sharding_info are totally separate. That would create a false impression that sharding info can be set per table. I didn't expect we want multiple sharding algorithms per node. If that's true then it changes my perspective a bit.

Reply:
done - sharding_info is in schema now.